### PR TITLE
Add fortcl submodule (#22)

### PR DESCRIPTION
**Pull request by arporter**

### CODE REVIEW

**Comment by arporter** [comment]
Hi @hiker, this one's ready to go if you wouldn't mind reviewing? Thanks.

File: finite_difference/src/Makefile [hunk 1]
```diff
@@ -81,19 +92,31 @@ SOURCES = $(BASE_SOURCES:%=$(ROOT)/%)
...
-	${AR} ${ARFLAGS} ${API_LIB} ${MODULES}
+# Create the archive. Note that we include the FortCL objects in this
+# archive but take care to exclude clfortran.o (as that only has
+# interface definitions, not implementation).
+${API_LIB}: fortcl ${MODULES}
+	${AR} ${ARFLAGS} ${API_LIB} ${FORTCL_DIR}/*.o ${MODULES}
+	${AR} d ${API_LIB} clfortran.o
 
+fortcl:
+	${MAKE} -C ${FORTCL_DIR} ${FORTCL_TARGET}
```
Comment (hiker) [inline finite_difference/src/Makefile]: While ``make F90=gfortran`` works as expected, just doing ``make`` does not (it somehow ends up with F90='c'.

Adding ``F90=$(F90)`` to the make call  seems to work fine.

File: finite_difference/src/Makefile [hunk 2]
```diff
@@ -81,19 +92,31 @@ SOURCES = $(BASE_SOURCES:%=$(ROOT)/%)
...
 
 install:
 	${MAKE} -C .. install
 clean:
 	rm -f *.o *.mod *.a
```
Comment (hiker) [inline finite_difference/src/Makefile:117]: Would be great to also clean fortcl in this case.

**Review by hiker — COMMENTED** [review]
I'll do a better review tomorrow, this was just what I noticed when trying a quick compile.

**Comment by arporter** [comment]
Thanks for those pointers. It happened that I had F90 defined in my shell so I didn't see that error.
Should be fixed now.

File: README.md [hunk 3]
```diff
@@ -25,3 +25,19 @@ The grid of points making up the simulation domain.
...
+
+TBD
+
+## OpenCL Support ##
+
+dl_esm_inf incorporates the FortCL library (github.com/stfc/FortCL) as
+a git sub-module. Obtaining the source of this submodule requires that
+you either supply the ``--recursive`` flag to ``git clone`` or that
+you subsequently do ``git submodule init`` and ``git submodule
+update``. By default a stub version of this library is compiled so
```
Comment (hiker) [inline README.md:39]: Thanks for the instructions!

File: finite_difference/src/Makefile [hunk 4]
```diff
@@ -25,7 +25,8 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #------------------------------------------------------------------------------
-# Author: A. R. Porter, STFC Daresbury Laboratory
+# Authors: A. R. Porter, STFC Daresbury Laboratory
+#          J. Henrichs, Australian Bureau of Meteorology
```
Comment (hiker) [inline finite_difference/src/Makefile:29]: Thanks for adding me :)

File: finite_difference/src/Makefile [hunk 5]
```diff
@@ -116,7 +141,7 @@ tile_mod.o: region_mod.o
 subdomain_mod.o: $(PARALLEL_MOD).o region_mod.o
 
 %.o: $(ROOT)/%.[Ff]90
-	$(F90) $(F90FLAGS) -c $<
+	$(F90) $(F90FLAGS) -I${FORTCL_DIR} -c $<
```
Comment (hiker) [inline finite_difference/src/Makefile:144]: I actually had to look it up, but indeed in make $() and ${} are the same ... I learned something :)

**Review by hiker — APPROVED** [review]

**Comment by hiker** [comment]
All issues address, library compiles fine.

**Pull request merged**